plugins/views/frontend/public/javascripts/countly.views.js

```diff
@@ -1,797 +1,438 @@
-                    self.drawGraph();
                 });
+            },
+            handleSelectionChange: function(selectedRows) {
 
-                var persistentSettings = countlyCommon.getPersistentSettings()['pageViewsItems_' + countlyCommon.ACTIVE_APP_ID] || [];
-                self.selectedViews = [];
-
-                for (var l in persistentSettings) {
-                    var current = persistentSettings[l];
-                    if (self.selectedViews.indexOf(current) < 0) {
-                        self.selectedViews.push(current);
-                    }
+                var selected = countlyCommon.getPersistentSettings()["pageViewsItems_" + countlyCommon.ACTIVE_APP_ID] || [];
+                var map = {};
+                for (var kz = 0; kz < selected.length; kz++) {
+                    map[selected[kz]] = true;
                 }
-                $("#view-metric-" + this.selectedMetric).parents(".big-numbers").addClass("active");
-                if (self.selectedViews.length > 0) {
-                    countlyViews.setSelectedViews(self.selectedViews);
-                    $.when(countlyViews.refresh()).then(function() {
-                        self.drawGraph();
-                    });
+                selected = Object.keys(map); //get distinct
+                if (selected.indexOf(selectedRows) === -1) {
+                    selected.push(selectedRows);
                 }
                 else {
-                    self.drawGraph();
-
+                    var index = selected.indexOf(selectedRows);
+                    selected.splice(index, 1);
                 }
-            }
+                var persistData = {};
+                persistData["pageViewsItems_" + countlyCommon.ACTIVE_APP_ID] = selected;
+                countlyCommon.setPersistentSettings(persistData);
 
-            if (!countlyAuth.validateUpdate(FEATURE_NAME)) {
-                $('#manage-views-button').hide();
-            }
-        },
-        drawGraph: function() {
-            var dp = [];
-            for (var i = 0; i < this.selectedViews.length; i++) {
-                var color = countlyCommon.GRAPH_COLORS[i];
-                var data = countlyViews.getChartData(this.selectedViews[i], this.selectedMetric, this.viewMap[this.selectedViews[i]], this.selectedSegment.segmentKey, this.selectedSegment.segmentValue).chartDP;
-                if (data) {
-                    for (var index = 0; index < data.length; index++) {
-                        var point = data[index];
-                        if (point && point.label) {
-                            var n = 25;
-                            point.label = point.label.substr(0, n) + '&hellip;';
-                        }
+                for (var k = 0; k < this.$refs.viewsTable.sourceRows.length; k++) {
+                    if (selected.indexOf(this.$refs.viewsTable.sourceRows[k]._id) > -1) {
+                        this.$refs.viewsTable.sourceRows[k].selected = true;
```

Comment:
It might be better to do this via table mutations (patch/unpatch API). But if it works, we can make an exception for this PR.